### PR TITLE
Improve I18n implementation in payment views.

### PR DESCRIPTION
**Pull request by Murph33**
This includes model name translations and model attribute translations in order to improve I18n use.

This is part of an ongoing effort as discussed in #735.

### CODE REVIEW

File: backend/app/views/spree/admin/payments/_form.html.erb [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
-      <label><%= Spree.t(:payment_method) %></label>
+      <label><%= Spree::PaymentMethod.model_name.human %></label>
       <ul>
         <% @payment_methods.each do |method| %>
-          <li data-id="<%= Spree.t(method.name, :scope => :payment_methods, :default => method.name).parameterize %>">
+          <li data-id="<%= Spree.t("#{method.name.downcase.tr(" ", "_")}").parameterize %>">
             <label data-hook="payment_method_field">
               <%= radio_button_tag 'payment[payment_method_id]', method.id, method == @payment_method, { class: "payment_methods_radios" } %>
-              <%= Spree.t(method.name, :scope => :payment_methods, :default => method.name) %>
+              <%= Spree.t("#{method.name.downcase.tr(" ", "_")}") %>
```
Comment (jhawthorn) [inline backend/app/views/spree/admin/payments/_form.html.erb]: I'm not sure why this change was made, the existing code looks preferable.

Comment (Murph33) [inline backend/app/views/spree/admin/payments/_form.html.erb]: There is no payment_methods scope in the base solidus dictionary and the default PaymentMethods are named `"Credit Card"`, `"Store Credit"`, and `"Check".  Only Check is close to the syntax we are using in the I18n dictionary for keys.  The current way it works just by abusing the functionality of default not throwing a translation missing when it's used.  This manipulation was the way I thought of actually using keys in the dictionary though I agree it isn't the cleanest approach.

Comment (jhawthorn) [inline backend/app/views/spree/admin/payments/_form.html.erb]: I don't think we should be translating these method names at all.

Comment (Murph33) [inline backend/app/views/spree/admin/payments/_form.html.erb]: That does make the most sense.  Can't see a good reason to manipulate something so we can translate `Credit Card` to `Credit Card` (in any language)

File: backend/app/views/spree/admin/payments/new.html.erb [hunk 2]
```diff
@@ -21,6 +21,6 @@
     </fieldset>
   <% end %>
 <% else %>
-  <%= Spree.t(:cannot_create_payment_without_payment_methods) %>
-  <%= link_to Spree.t(:please_define_payment_methods), admin_payment_methods_url %>
+  <%= Spree.t(:cannot_create_payment_without_payment_methods_html,
+              link:(link_to Spree.t(:cannot_create_payment_link), admin_payment_methods_url)) %>
```
Comment (jhawthorn) [inline backend/app/views/spree/admin/payments/new.html.erb]: Was this necessary to combine?

If so this line should be:
```
link: link_to(Spree.t(:cannot_create_payment_link), admin_payment_methods_url))
```

(Using the parens for the method arguments rather than around the whole method call)

Comment (Murph33) [inline backend/app/views/spree/admin/payments/new.html.erb]: I felt that by combining it we don't take any assumptions about the grammar of a language someone might be using.  Simply shows that we want to provide a link to creating a new payment method and a note explaining that there currently are none.

**Comment by cbrunsdon** [comment]
:+1: , thanks @Murph33

**Comment by jhawthorn** [comment]
:+1:
